mmv1/products/redis/api.yaml

```diff
@@ -143,6 +143,125 @@ objects:
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::Enum
+                  name: 'day'
+                  required: true
+                  description: |
+                    Required. The day of week that maintenance updates occur.
+
+                    - DAY_OF_WEEK_UNSPECIFIED: The day of the week is unspecified.
+                    - MONDAY: Monday
+                    - TUESDAY: Tuesday
+                    - WEDNESDAY: Wednesday
+                    - THURSDAY: Thursday
+                    - FRIDAY: Friday
+                    - SATURDAY: Saturday
+                    - SUNDAY: Sunday
+                  values:
+                    - :DAY_OF_WEEK_UNSPECIFIED
+                    - :MONDAY
+                    - :TUESDAY
+                    - :WEDNESDAY
+                    - :THURSDAY
+                    - :FRIDAY
+                    - :SATURDAY
+                    - :SUNDAY
+                - !ruby/object:Api::Type::String
+                  name: 'duration'
+                  output: true
+                  description: |
+                    Output only. Duration of the maintenance window.
+                    The current window is fixed at 1 hour.
+                    A duration in seconds with up to nine fractional digits,
+                    terminated by 's'. Example: "3.5s".
+                - !ruby/object:Api::Type::NestedObject
+                  name: 'startTime'
+                  required: true
+                  allow_empty_object: true
+                  send_empty_value: true
+                  description: |
+                    Required. Start time of the window in UTC time.
+                  properties:
+                    - !ruby/object:Api::Type::Integer
+                      name: 'hours'
+                      default_value: 0
+                      description: |
+                        Hours of day in 24 hour format. Should be from 0 to 23.
+                        An API may choose to allow the value "24:00:00" for scenarios like business closing time.
+                    - !ruby/object:Api::Type::Integer
+                      name: 'minutes'
+                      default_value: 0
+                      description: |
+                        Minutes of hour of day. Must be from 0 to 59.
+                    - !ruby/object:Api::Type::Integer
+                      name: 'seconds'
+                      default_value: 0
+                      description: |
+                        Seconds of minutes of the time. Must normally be from 0 to 59.
+                        An API may allow the value 60 if it allows leap-seconds.
+                    - !ruby/object:Api::Type::Integer
+                      name: 'nanos'
+                      default_value: 0
+                      description: |
+                        Fractions of seconds in nanoseconds. Must be from 0 to 999,999,999.
```

Comment:
Are these fields required? Could one be required? otherwise someone could provide an empty startTime block

Reply:
Actually, API does accept an empty startTime block (and sets all values to 0 - midnight) - so I mapped API behavior exactly.
If you'd like me to set any of them as required anyway, I can do it - just let me know.